src/wp-includes/class-wp-scripts.php

```diff
@@ -445,31 +471,99 @@ public function add_inline_script( $handle, $data, $position = 'after' ) {
 	 * Prints inline scripts registered for a specific handle.
 	 *
 	 * @since 4.5.0
+	 * @deprecated 6.3.0 Use methods get_inline_script_tag() or get_inline_script_data() instead.
 	 *
-	 * @param string $handle   Name of the script to add the inline script to.
+	 * @param string $handle   Name of the script to print inline scripts for.
 	 *                         Must be lowercase.
 	 * @param string $position Optional. Whether to add the inline script
 	 *                         before the handle or after. Default 'after'.
-	 * @param bool   $display  Optional. Whether to print the script
-	 *                         instead of just returning it. Default true.
-	 * @return string|false Script on success, false otherwise.
+	 * @param bool   $display  Optional. Whether to print the script tag
+	 *                         instead of just returning the script data. Default true.
+	 * @return string|false Script data on success, false otherwise.
 	 */
 	public function print_inline_script( $handle, $position = 'after', $display = true ) {
-		$output = $this->get_data( $handle, $position );
+		_deprecated_function( __METHOD__, '6.3.0', 'WP_Scripts::get_inline_script_data() or WP_Scripts::get_inline_script_tag()' );
```

Comment:
Not sure about the usage of _multiple_ functions with " or " here, I haven't seen that pattern anywhere in core before. The more precise replacement is `get_inline_script_tag()`, so I'd suggest we provide only that.
```suggestion
		_deprecated_function( __METHOD__, '6.3.0', 'WP_Scripts::get_inline_script_tag()' );
```

Reply:
Which function you use depends on whether you are setting the `$display` parameter to `true` or not. There are already a couple of places in core where multiple functions are cited in the deprecation message, so I think this is fine.

Examples:
* https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-admin/includes/class-wp-filesystem-base.php#L138
* https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/class-wpdb.php#L1375